alchemy/src/apply.ts

```diff
@@ -161,34 +164,67 @@ async function _apply<Out extends Resource>(
       seq: resource[ResourceSeq],
       props: state.oldProps,
       state,
-      replace: () => {
+      replace: async () => {
         if (isReplaced) {
           logger.warn(
             `Resource ${resource[ResourceKind]} ${resource[ResourceFQN]} is already marked as REPLACE`,
           );
           return;
         }
+
         isReplaced = true;
+        throw new ReplacedSignal();
       },
     });
 
-    const output = await alchemy.run(
-      resource[ResourceID],
-      {
-        isResource: true,
-        parent: scope,
-      },
-      async (scope) => {
-        options?.resolveInnerScope?.(scope);
-        return provider.handler.bind(ctx)(resource[ResourceID], props);
-      },
-    );
+    let output: Resource<string>;
+    try {
+      output = await alchemy.run(
+        resource[ResourceID],
+        {
+          isResource: true,
+          parent: scope,
+        },
+        async (scope) => {
+          options?.resolveInnerScope?.(scope);
+          return provider.handler.bind(ctx)(resource[ResourceID], props);
+        },
+      );
+    } catch (error) {
+      if (error instanceof ReplacedSignal) {
+        ctx.phase = "create";
+
+        if (
+          Scope.current.children.get(resource[ResourceID])?.children.size! > 0
+        ) {
```

Comment:
Might be safer to just reference `scope` instead of `Scope.current`

Reply:
Can probably also link to an issue tracking enhancing REPLACE to support children